pkg/networking/cilium/templater.go

```diff
@@ -24,18 +23,19 @@ const (
 	defaultBackOffPeriod = 5 * time.Second
 )
 
-type Helm interface {
-	Template(ctx context.Context, ociURI, version, namespace string, values interface{}, kubeVersion string) ([]byte, error)
-	RegistryLogin(ctx context.Context, registry, username, password string) error
+// HelmClientFactory provides a helm client for a cluster.
+type HelmClientFactory interface {
+	Get(ctx context.Context, clus *anywherev1.Cluster) (helm.RegistryClient, error)
```

Comment:
why can't we just return `helm.Client` here and avoid the second interface?

Reply:
We can. I mentioned this above as well, so I mainly created this interface to scope down the methods available to the client returned by the factory, as only `RegistryLogin` and `Template` is called on it at some point.

ExecuteableClient is more general interface that has more methods due to the other consumers of BuildHelmExecutable

Though, we can remove it if it's not necessary